content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,250 @@
+
+### Referral hand image
+![hand](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/hand.jpg)
+
+*[Image Source: Mediapipe](https://google.github.io/mediapipe/solutions/hands.html)*
+
+The above image shows the numbers of the points that mediapipe uses to refer to different points of the hand. This tutorial will use point `4` and point `8` which are the thumb and the index finger respectively.
+
+### Setting up
+First of all, we will prepare our workspace. Launch the Pycharm app. Click on the create a new project.
+
+![Pycharm](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/pycharm.png)
+
+Click on the create button on the window that appears next.
+
+We need to install the libraries we need to use in our project.
+
+`numpy` will help us work with arrays. To install it, open the terminal and run the following command: 
+
+```bash
+pip install numpy
+```
+Repeat the same process for the other libraries.
+
+```bash
+pip install opencv-python
+```
+This library is imported as `cv2`. We will use it to capture an image using the webcam and convert the image to `RGB`.
+
+```bash
+pip install mediapipe
+```
+This is an open-source library developed by Google. It's used for both face and gesture recognition. For this tutorial, we'll use it for hand gesture recognition.
+
+```bash
+pip install pycaw
+```
+We'll need this library to access the device's speaker and its master volume.
+
+```bash
+pip install python-math
+```
+We'll use this library to find the distance between point number `4` (the thumb) and point number `8` (the index finger) using hypotenuse.
+
+```bash
+pip install gpib-ctypes, comtypes
+```
+`pycaw` depends on these two libraries. `Ctypes` providea `C` language compatible data types. `Comtypes` bases on the `ctypes` **FFI**(Foreign Function Interface) library.
+
+Now let us jump into coding. This is where the most fun is as we get hands-on.
+
+```python
+import cv2
+import mediapipe as mp
+from math import hypot
+from ctypes import cast, POINTER
+from comtypes import CLSCTX_ALL
+from pycaw.pycaw import AudioUtilities, IAudioEndpointVolume
+import numpy as np
+```
+In the above code segment, we are importing each library we did discuss into our project.
+
+```python
+cap = cv2.VideoCapture(0) 
+```
+We are getting the video input from our computer's primary camera. If you are using any other camera, replace the number `0` with that of the camera you are using.
+
+### Detecting, initializing, and configuring the hands
+```python
+mpHands = mp.solutions.hands
+hands = mpHands.Hands()
+mpDraw = mp.solutions.drawing_utils
+```
+- `mp.solutions.hands`: We are calling on the `mediapipe` hand module to detect the hands from the video input we did get.
+- `mpHands.Hands()`: This completes the initialization and configuration of the hands we did detect.

```

Comment:
I am seeing several sentences like this which are somehow vague and grammatically incorrect. Please check on them. For instance, instead of `...we did detect` you can say `...we detected`. Some statements are in passive voices. Change them to active voice.

Also, check your dictum. Make use of simple words that best describe what you want to get across.